tests/unit/s2n_client_key_share_extension_test.c

```diff
@@ -165,11 +164,40 @@ int main(int argc, char **argv)
     /* Test s2n_client_key_share_extension.recv */
     {
         /* Test that s2n_client_key_share_extension.recv is a no-op
-         * if tls1.3 not enabled */
+         * if tls1.3 not enabled AND in use  */
```

Comment:
When does the case of tls1.3 not enabled but in use happen?

Reply:
In the integration tests, it came up for a server with TLS1.3 enabled but only using TLS1.2 cipher suites. In that case, we fall back to TLS1.2: https://github.com/awslabs/s2n/blob/ecbf2b17e74a6b49b48dc261c91e2670f4819838/tls/s2n_client_hello.c#L304-L308